feat(settings): Adjust settings promo banner for mobile

vpomerleau opened the pull request:
## Because

* We want to match designs for mobile view

## This pull request

* Adjust styling of the NotificationPromoBanner component to support mobile view
* Adjust styling of PageSettings to use a gap between components (removes needs for setting external margins in section components
* Remove stacked banner story, the spacing between the banners is now set at the PageSettings level
* Adds story to PageSettings to show NotificationPromoBanner in context
* Small drive-by tweaks for ltr/rtl

## Issue that this pull request solves

Closes: #FXA-10405

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Will add storybook links when deployed

![image](https://github.com/user-attachments/assets/15660029-61b7-440d-83b9-52f9b0641659)

![image](https://github.com/user-attachments/assets/994c702b-5b23-4cad-bb3e-77d3674f7c5f)

![image](https://github.com/user-attachments/assets/355164d0-e195-4968-8a4a-6c1944eb8e0b)

![image](https://github.com/user-attachments/assets/e02f0cfa-66ea-41d3-8cd1-e4c4fa3a8585)


## Other information (Optional)

Any other information that is important to this pull request.

vpomerleau commented on packages/fxa-settings/src/components/NotificationPromoBanner/index.stories.tsx:
```diff
@@ -48,10 +48,3 @@ export const NotVisible = () => (
   ></NotificationPromoBanner>
 );
-
-export const Stacked = () => (
```
Spacing between sections/banners/etc. now controlled at the PageSettings level, no need for this story.

vpomerleau commented on packages/fxa-settings/src/components/Settings/ConnectedServices/index.tsx:
```diff
@@ -193,12 +193,11 @@ export const ConnectedServices = forwardRef<HTMLDivElement>((_, ref) => {
       id="connected-services-section"
       {...{ ref }}
     >
-      <h2 className="font-header font-bold mobileLandscape:ltr:ml-6 mobileLandscape:rtl:ml-6 ltr:ml-4 rtl:mr-4 mb-4 relative">
```
Using logical properties for LTR/RTL + fixing an error where rtl needs a margin on the right instead of left

vpomerleau commented on packages/fxa-settings/src/components/Settings/PageSettings/index.tsx:
```diff
@@ -96,7 +96,7 @@ export const PageSettings = (_: RouteComponentProps) => {
         />
       </div>
-      <div className="flex-7 max-w-full">
+      <div className="flex flex-col flex-7 max-w-full gap-8 mt-10">
```
Spacing between sections now controlled here instead of in each section

vpomerleau reviewed (commented):
Some self-review comments.

@bcolsson I tagged you for review even though there are no ftl changes since you flagged potential issues with German. Could you take a peek and see if it's resolved with these tweaks? Thanks!

vbudhram reviewed (commented):
@vpomerleau Looks great in mobile! Maybe a small tweak in desktop view is to make the `Create` button smaller.

Current figma looks like this

<img width="303" alt="Screenshot 2024-09-11 at 2 50 05 PM" src="https://github.com/user-attachments/assets/97802891-ba01-4730-bd7f-d01d126433f1">

bcolsson reviewed (approved):
Thanks for flagging, yes tested this on storybook with some German text (including the term Kontowiederherstellungsschlüssel) and it seems to be able to handle mobile views decently well.

LZoog reviewed (approved):
I agree with Vijay about the button size in desktop. I think one other thing in desktop could be to increase the padding on the top/bottom by around by 2px, add more space on the left/right sides, especially the space between the "Create" and "close" button, decrease spacing between the bolded text and text, and increase the SVG size? (Sorry if it's scope creep since your PR is mobile. The only thing I'd insist on would be the space between close and create buttons.) LGTM in mobile so r+!

<img width="273" alt="image" src="https://github.com/user-attachments/assets/fc66daee-9232-4c5d-8071-bc46eef1f774">
<img width="207" alt="image" src="https://github.com/user-attachments/assets/1a783f71-87d7-4c84-8c90-5dbb436c0f77">
^Figma, vs Storybook:
<img width="242" alt="image" src="https://github.com/user-attachments/assets/538451a3-8f72-4d27-9ba6-00683d46742d">
<img width="208" alt="image" src="https://github.com/user-attachments/assets/56dc071a-3dda-45ed-b229-c47d6c902d91">

vpomerleau commented:
Thanks all! I've adjusted based on your comments and some discussion with Eduardo on Slack. Couple of additional changes not mentioned here yet:
- remove the box shadow for better contrast
- align the banner CTA with CTAs in the sections below
- tweak padding/spacing between close icon and CTA to provide best possible target size while avoiding overlap between buttons
- added a transparent border on the banner for high contrast mode
